x-pack/plugins/stack_connectors/server/connector_types/gen_ai/lib/utils.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AxiosRequestHeaders, AxiosResponse, ResponseType } from 'axios';
+import { IncomingMessage } from 'http';
+import { OpenAiProviderType } from '../../../../common/gen_ai/constants';
+import {
+  sanitizeRequest as openAiSanitizeRequest,
+  getRequestWithStreamOption as openAiGetRequestWithStreamOption,
+} from './openai_utils';
+import {
+  sanitizeRequest as azureAiSanitizeRequest,
+  getRequestWithStreamOption as azureAiGetRequestWithStreamOption,
+} from './azure_openai_utils';
+
+export const sanitizeRequest = (provider: string, url: string, body: string): string => {
+  switch (provider) {
+    case OpenAiProviderType.OpenAi:
+      return openAiSanitizeRequest(url, body);
+    case OpenAiProviderType.AzureAi:
+      return azureAiSanitizeRequest(url, body);
+    default:
+      return body;
+  }
+};
+
+export const getRequestWithStreamOption = (
+  provider: string,
+  url: string,
+  body: string,
+  stream: boolean
+): string => {
+  switch (provider) {
+    case OpenAiProviderType.OpenAi:
+      return openAiGetRequestWithStreamOption(url, body, stream);
+    case OpenAiProviderType.AzureAi:
+      return azureAiGetRequestWithStreamOption(url, body, stream);
+    default:
+      return body;
+  }
+};
+
+export const getAxiosOptions = (
+  provider: string,
+  apiKey: string,
+  stream: boolean
+): { headers: AxiosRequestHeaders; responseType?: ResponseType } => {
+  const responseType = stream ? { responseType: 'stream' as ResponseType } : {};
+  switch (provider) {
+    case OpenAiProviderType.OpenAi:
+      return {
+        headers: { Authorization: `Bearer ${apiKey}`, ['content-type']: 'application/json' },
+        ...responseType,
+      };
+    case OpenAiProviderType.AzureAi:
+      return {
+        headers: { ['api-key']: apiKey, ['content-type']: 'application/json' },
+        ...responseType,
+      };
+    default:
+      return { headers: {} };
+  }
+};
+
+export const pipeStreamingResponse = (response: AxiosResponse<IncomingMessage>) => {
+  response.data.headers = {
+    ['Content-Type']: 'dont-compress-this',
```

Comment:
are we leaving this in?  It can't be copied to the eventual outgoing request, I assume, not sure if it's a special directive to the core http framework, or what.  Also, not sure that these sorts of things _can't_ be compressed, but there do tend to be issues w/compression and streaming APIs generally.

Reply:
Iirc, it will be compressed by hapi if it recognizes the content type. A bit of a hack but it works. Maybe there's a less quirky string that we can use though.

Reply:
Slack thread: https://elastic.slack.com/archives/C5TQ33ND8/p1685452168698039?thread_ts=1685076705.291449&cid=C5TQ33ND8

Reply:
Heh, cool.  Let's add a comment in the code, as this will confuse someone in the future!

Reply:
Added in https://github.com/elastic/kibana/pull/161676/commits/ecec58cddc730746027bf5d3b4d6e60adca027b3